lib/chef/resource/aws_rds_instance.rb

```diff
@@ -28,12 +28,21 @@ class Chef::Resource::AwsRdsInstance < Chef::Provisioning::AWSDriver::AWSRDSReso
   attribute :additional_options, kind_of: Hash, default: {}
 
   def aws_object
-    res = driver.rds.instances[name]
-    if res.exists? && ! ['deleted', 'deleting'].include?(res.status)
-      res
-    else
-      nil
-    end
+    result = self.driver.rds_resource.db_instance(name)
+    return nil unless result && result.db_instance_status != 'deleting'
+    result
+  rescue ::Aws::RDS::Errors::DBInstanceNotFound
+    nil
+  end
+
+  def endpoint
```

Comment:
I should add that ```endpoint``` and ```db_instance_status``` were added because I needed them, notably for getting the endpoint value so that clients can connect to the RDS instance provisioned. I understand this may have some ramifications for resource cloning, but I'm not sure exactly what, so any insight will be appreciated.